packages/keystone/sliceSchema.js

```diff
@@ -0,0 +1,484 @@
+     * @param field schema field object
+     * @returns {Set<string>} used enums
+     */
+    findEnumsFromField (field) {
+        const enums = new Set()
+        let fieldType = field.type
+        let fieldKind = fieldType.kind
+        while (fieldType.ofType) {
+            fieldType = fieldType.ofType
+            fieldKind = fieldType.kind
+        }
+        if (fieldKind === 'ENUM') {
+            enums.add(fieldType.name)
+        }
+        const args = get(field, 'args', []) || []
+        for (const arg of args) {
+            const argEnums = this.findEnumsFromField(arg)
+            for (const enumName of argEnums) {
+                enums.add(enumName)
+            }
+        }
+
+        return enums
+    }
+
+    getTypeByName (name) {
+        if (!(name in this.typesByNames)) {
+            throw new Error(`Expected ${name} type to be in original schema types, but it was not found (or no longer exist)`)
+        }
+        return this.typesByNames[name]
+    }
+}
+
+/**
+ * Configuration of GraphQL schema slices, that should recursively be extracted from GraphQL schema in question
+ *
+ * @type SliceConfig
+ * @property {String} name - name of GraphQL type
+ * @property {String[]} operations - list of GraphQL CRUD operations
+ * @property {String} fields - space-separated list of GraphQL type fields
+ * @example { name: 'AcquiringIntegrationContext', operations: ['read', 'update'], fields: '{ id name organization { id name } }' }
+ */
+
+/**
+ * Takes part from provided GraphQL schema, referenced by provided slicing configuration.
+ * Slicing configuration specifies GraphQL types with operations and fields, that should be extracted recursively from the schema.
+ *
+ * Main usage is to extract part of GraphQL schema needed for microservice in question from "condo" GraphQL schema
+ * Extracted part will have executor targeted to "condo" endpoint
+ * Extracted part will be stitched with schema of the microservice
+ * This way a client will get a uniform GraphQL API, that have different executors under the hood
+ *
+ * @param schema
+ * @param {SliceConfig} sliceConfig
+ * @example
+ * const condoSchema = await loadSchema('../condo/schema.graphql', {
+ *     loaders: [new GraphQLFileLoader()],
+ * })
+ * const sliceConfig = { name: 'AcquiringIntegrationContext', operations: ['read', 'update'], fields: '{ id name organization { id name } }' }
+ * const sliceIntrospection = await sliceSchema(condoSchema, sliceConfig)
+ * const slicePrint = printSchema(sliceIntrospection)
+ * fs.writeFileSync('./condoSchema.graphql', slicePrint)
+ */
+async function sliceSchema (schema, sliceConfig) {
+    const schemaIntrospection = await introspectionFromSchema(schema)
+    const cleaner = new CondoSchemaCleaner(sliceConfig, schemaIntrospection)
+    cleaner.cleanCondoSchema()
+    const cleanedIntrospection = cleaner.getCleanedIntrospection()
+    return buildClientSchema(cleanedIntrospection)
+}
+
+module.exports = {
+    sliceSchema,
```

Comment:
I couldn't find any usages of this function. Should it be a part of this PR?

Reply:
Thank you for spotting this.
It has been mistakenly slipped into this PR.
Removed